test/app/model/ChessEngine.spec.ts

```diff
@@ -445,3 +444,187 @@ describe('Return player moves without those that cause his king to check', () =>
     );
   });
 });
+
+describe('Castling can be done only if neither king nor rook has moved and none of square which the king is passing by is under attack and no piece between the king and the rook', () => {
+  const whiteKing = new King(Side.WHITE);
+  const whiteRookA1 = new Rook(Side.WHITE);
+  const whiteRookH1 = new Rook(Side.WHITE);
+  const whiteBishop = new Bishop(Side.WHITE);
+  const blackKing = new King(Side.BLACK);
+  const blackRook = new Rook(Side.BLACK);
+  const blackKnight = new Knight(Side.BLACK);
+  const blackPawn = new Pawn(Side.BLACK);
+
+  it('When all conditions are fulfilled, then short castling can be done', () => {
```

Comment:
When all conditions are fulfilled,  - tego bym chyba nie powtarzal :) Mozesz to potraktowac bardziej jako "scenariusze testowe" i dac nazwe scenariusza. Bo na gorze nie zaczelac od Given, tylko od opisu reguły.